syntax = "proto3";

package transport_serialize;

import "map_renderer.proto";
import "graph.proto";
import "transport_router.proto";

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    uint32 id = 1;
    string name = 2;
    Coordinates coordinates = 3;
}

message Bus {
    uint32 id = 1;
    string name = 2;
    repeated uint32 stop_ids = 3;
    bool is_roundtrip = 4;
}

message StopToBuses {
    uint32 id = 1;
    repeated uint32 bus_ids = 2;
}

message Distance {
    uint32 from = 1;
    uint32 to = 2;
    int32 distance = 3;
}

message GeoDistance {
    uint32 from = 1;
    uint32 to = 2;
    double distance = 3;
}

message BusToTotal {
    uint32 id = 1;
    int32 distance = 2;
    double geo_distance = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated StopToBuses stop_to_buses = 3;
    repeated Distance distances = 4;
    repeated GeoDistance geo_distances = 5;
    repeated BusToTotal bus_id_to_total_distances = 6;
}

message SaveData {
    TransportCatalogue transport_catalogue = 1;
    renderer_serialize.RenderSettings render_settings = 2;
    router_serialize.Graph graph = 3;
    router_serialize.RouterSettings router_data = 4;
}